### ISSUE TITLE
v3.0.2-alpha.0 introduces a bug that causes modal to not dismiss properly if dismiss pressable is inside modal

### ISSUE THREAD

**Issue by mr-moto** [gorhom/react-native-bottom-sheet#171]
# Bug
`v3.0.20alpha` introduces a modal dismiss bug.
If a dismiss function is called from within the modal ( eg button inside modal that calls `dismiss()` ), `handleOnChange` will not be called and will prevent the modal from presenting once closed. similar to #168.

https://github.com/gorhom/react-native-bottom-sheet/blob/1257c462f68f4a0550211dddaffeecdd7d7c56a3/src/components/bottomSheet/BottomSheet.tsx#L573-L574
the state of `_contentGestureState ` is 2 (onStart) which does not equal `END` or `UNDETERMINED`
 

## Environment info

| Library                         | Version |
| ------------------------------- | ------- |
| @gorhom/bottom-sheet         | 3.0.2-alpha.3   |
| react-native                    | 0.63.4   |
| react-native-reanimated         | 2.0.0-rc.1  |
| react-native-gesture-handler    | 1.9.0   |

## Steps To Reproduce
1. run the modal simple example
2. add a button inside a modal 
```
<TouchableOpacity onPress={()=>modalRef.current.dismiss()}><Text>Dismiss by ref</Text></TouchableOpacity>
<TouchableOpacity onPress={dismissModal}><Text>Dismiss by calling function</Text></TouchableOpacity>
<TouchableOpacity onPress={()=>dismiss('modalName')}><Text>Dismiss by hook</Text></TouchableOpacity>
<TouchableOpacity onPress={dismissAll}><Text>Dismiss All by hook</Text></TouchableOpacity>
```
3. present -> dismiss using one of the buttons inside the modal -> present ( does not present )

Describe what you expected to happen:

1. present modal after dismissing

**Comment by mr-moto** [comment]
https://github.com/gorhom/react-native-bottom-sheet/blob/1257c462f68f4a0550211dddaffeecdd7d7c56a3/src/hooks/useInteractivePanGestureHandler.ts#L35
the `onActive` and `onEnd` is not triggered in the gesture handler. only `onStart` gets triggered which sets the state to 2 ( which is the same behavior as 3.0.1 ) so I assume we just need to remove that part of the `if` statement or change it to `_contentGestureState === State.START`

**Comment by gorhom** [comment]
thanks @mr-moto for spotting this bug ! indeed it is an issue with `useInteractivePanGestureHandler`!

submitting a pr shortly :)

**Comment by gorhom** [comment]
this should be resolved with `v3.1.0` 🎉
